### PR TITLE
crimson/osd/pg: fix wrong lambda capture of transactions in `PG::submit_error_log()`

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/67450
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
The following tests FAILED:
	  4 - run-tox-mgr (Failed)
	 22 - run-tox-cephadm (Failed)

**Comment by xxhdx1985126** [comment]
jenkins test make check

**Comment by xxhdx1985126** [comment]
jenkins test api

File: src/crimson/osd/pg.cc [hunk 1]
```diff
@@ -1148,7 +1148,7 @@ seastar::future<std::optional<eversion_t>> PG::submit_error_log(
       return shard_services.send_to_osd(peer.osd,
                                         std::move(log_m),
                                         get_osdmap_epoch());
-    }).then([this, waiting_on, t=std::move(t), rep_tid] () mutable {
```
Comment (cyx1231st) [inline src/crimson/osd/pg.cc:1151]: Looks `do_transaction(t)` is the only user, so `t` doesn't need to be wrapped by `do_with()` to be moved there?

Comment (xxhdx1985126) [reply src/crimson/osd/pg.cc]: Yeah, done, please take a look again.

**Review by cyx1231st — APPROVED** [review]

**Review by Matan-B — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
Missed this PR, will test in next batch

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
https://shaman.ceph.com/builds/ceph/wip-matanb-crimson-testing-sep1/

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-09-02_14:15:24-crimson-rados-wip-matanb-crimson-testing-sep1-distro-crimson-smithi/

**Comment by Matan-B** [comment]
jenkins test make check

**Pull request merged**